lib/phoenix_live_view/html_engine.ex

```diff
@@ -747,6 +771,13 @@ defmodule Phoenix.LiveView.HTMLEngine do
     end
   end
 
+  defp store_component_call(module, component, attrs, file, line) do
+    if Module.open?(module) do
+      call = %{component: component, attrs: attrs, file: file, line: line}
```

Comment:
For the attrs, I think we can convert them to atoms at this point. This will simplify the checking later on. Also, I don't think we need to pass the attribute values, so don't pass them if you do it today.

Btw, what happens if you have a required attribute but you are passing it dynamically? Such as:

```elixir
  <Foo.bar {[some_required_attribute: "value"]}>
```

Should we disable the required validation if dynamic attributes are given? Or should we force required attributes to be given explicitly?

Reply:
In Surface, we disable the validation for required props whenever dynamic attributes are passed as there's no way to consistently perform the check.